src/mlpack/methods/cf/decomposition_policies/randomized_svd_method.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file randomized_svd_method.hpp
+ * @author Haritha Nair
+ *
+ * Implementation of the randomized svd method for use in the Principal
+ * Components Analysis method.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_CF_DECOMPOSITION_POLICIES_RANDOMIZED_SVD_METHOD_HPP
+#define MLPACK_METHODS_CF_DECOMPOSITION_POLICIES_RANDOMIZED_SVD_METHOD_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/randomized_svd/randomized_svd.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * Implementation of the randomized SVD policy.
+ */
+class RandomizedSVDPolicy
+{
+ public:
+  /**
+   * Use randomized SVD method to perform collaborative filtering
+   *
+   * @param iteratedPower Size of the normalized power iterations
+   *        (Default: rank + 2).
+   * @param maxIterations Number of iterations for the power method
+   *        (Default: 2).
+   */
+  RandomizedSVDPolicy(const size_t iteratedPower = 0,
+                      const size_t maxIterations = 2) :
+      iteratedPower(iteratedPower),
+      maxIterations(maxIterations)
+  {
+    /* Nothing to do here */
+  }
+
+  /**
+   * Apply Collaborative Filtering to the provided data set using the
+   * randomized SVD.
+   *
+   * @param data Data matrix: dense matrix (coordinate lists) 
+   *    or sparse matrix(cleaned).
+   * @param cleanedData item user table in form of sparse matrix.
+   * @param rank Rank parameter for matrix factorization.
+   * @param w First matrix formed after decomposition.
+   * @param h Second matrix formed after decomposition.
+   * @param maxIterations Maximum number of iterations.
+   * @param minResidue Residue required to terminate.
+   * @param mit Whether to terminate only when maxIterations is reached.
+   */
+  template<typename MatType>
+  void Apply(MatType const& /* data */,
+             const arma::sp_mat& cleanedData,
+             const size_t rank,
+             arma::mat& w,
+             arma::mat& h,
+             const size_t maxIterations,
+             const double /* minResidue */,
+             const bool /* mit */)
+  {
+    arma::vec sigma;
+
+    // Do singular value decomposition using the randomized SVD algorithm.
+    svd::RandomizedSVD rsvd(rank+2, maxIterations);
```

Comment:
I think this should be ``svd::RandomizedSVD rsvd(ìterationPower, maxIterations)``.